docs/discovery.md

```diff
@@ -155,3 +102,58 @@ class MyClass
     }
 }
 ```
+
+
+## Integrating your own implementation with the discovery mechanism
+
+The `php-http/discovery` has built-in support for some implementations. To use a different implementation or override 
+the default when several implementations are available in your codebase, you can register a class explicitly with the 
```

Comment:
I prefer writting markdown files continously without character limit per line. IDEs can break lines automatically when showing text. It is not code, readability is fine this way as well IMO.